packages/api/src/cli/storage.ts

```diff
@@ -0,0 +1,297 @@
+import type { OASDocument } from 'oas/@types/rmoas.types';
+
+import ssri from 'ssri';
+import fs from 'fs';
+import path from 'path';
+import makeDir from 'make-dir';
+
+import { PACKAGE_VERSION } from '../packageInfo';
+
+import Fetcher from '../fetcher';
+
+export default class Storage {
+  static dir: string;
+
+  static lockfile: false | Lockfile;
+
+  /**
+   * This is the original source that the file came from (relative/absolute file path, URL, ReadMe
+   * registry UUID, etc.).
+   */
+  source: string;
+
+  identifier: string;
+
+  fetcher: Fetcher;
+
+  constructor(source: string, identifier?: string) {
+    Storage.setStorageDir();
+
+    this.fetcher = new Fetcher(source);
+
+    this.source = source;
+    this.identifier = identifier;
+
+    // This should default to false so we have awareness if we've looked at the lockfile yet.
+    Storage.lockfile = false;
+  }
+
+  static getLockfilePath() {
+    return path.join(Storage.dir, 'api.json');
+  }
+
+  static getAPIsDir() {
+    return path.join(Storage.dir, 'apis');
+  }
+
+  static setStorageDir(dir?: string) {
+    if (dir) {
+      Storage.dir = dir;
+      return;
+    } else if (Storage.dir) {
+      // If we already have a storage dir set and aren't explicitly it to something new then we
+      // shouldn't overwrite what we've already got.
+      return;
+    }
+
+    Storage.dir = makeDir.sync(path.join(process.cwd(), '.api'));
+
+    makeDir.sync(Storage.getAPIsDir());
+  }
+
+  /**
+   * Reset the state of the entire storage system.
+   *
+   * This will completely destroy the contents of the `.api/` directory!
+   */
+  static async reset() {
+    if (Storage.getLockfilePath()) {
+      await fs.promises.writeFile(Storage.getLockfilePath(), JSON.stringify(Storage.getDefaultLockfile(), null, 2));
+    }
+
+    if (Storage.getAPIsDir()) {
+      await fs.promises.rm(Storage.getAPIsDir(), { recursive: true });
+      await fs.promises.mkdir(Storage.getAPIsDir(), { recursive: true });
+    }
+  }
+
+  static getDefaultLockfile(): Lockfile {
+    return {
+      version: '1.0',
+      apis: [],
+    };
+  }
+
+  static generateIntegrityHash(definition: OASDocument) {
+    return ssri
```

Comment:
[ssri](https://npm.im/ssri) is the same library that NPM uses to generate integrity hashes in `package-lock.json` files.